pkgs/development/compilers/hare/hare/default.nix

```diff
@@ -51,29 +55,25 @@ stdenv.mkDerivation (finalAttrs: {
         if stdenv.isLinux then "linux"
         else if stdenv.isFreeBSD then "freebsd"
         else "unsupported";
-      hareflags = "";
-      config-file = substituteAll {
-        src = ./config-template.mk;
-        inherit arch platform hareflags;
-      };
     in
     ''
       runHook preConfigure
 
-      export HARECACHE="$NIX_BUILD_TOP/.harecache"
-      export BINOUT="$NIX_BUILD_TOP/.bin"
-      cat ${config-file} > config.mk
+      cp config.example.mk config.mk
+      makeFlagsArray+=(
+        PREFIX="${builtins.placeholder "out"}"
+        HARECACHE="$(mktemp -d --tmpdir harecache.XXXXXXXX)"
+        BINOUT="$(mktemp -d --tmpdir bin.XXXXXXXX)"
+        PLATFORM="${platform}"
+        ARCH="${arch}"
+      )
```

Comment:
Those don't work in makeFlags?

Reply:
With the exception of `HARECACHE` and `BINOUT`, they would. I've put them all together in `makeFlagsArray` so that the make flags are not split between `makeFlags` and `makeFlagsArray`.

But if it the ones that work on `makeFlags` should necessarily be in it, I can move them there.

Reply:
A note: github's `outdated` warning is because the file path has changed from `pkgs/development/compilers/hare/hare/default.nix` to `pkgs/development/compilers/hare/default.nix`. The `makeFlagsArray` has not changed.